File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.py
def solution(land):
    answer = 0

    # 이전 행의 자기 열 제외 max값 더하기
    for row in range(1, len(land)):
        for col in range(4):
            land[row][col] += max(land[row - 1][:col] + land[row - 1][col + 1:])
    
    return max(land[-1])

File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.js
function solution(land) {
    var answer = 0;

    for (let i = 1; i < land.length; i++) {
        const prev = land[i - 1];
        for (let j = 0; j < land[0].length; j++) {
            const targetArr = prev.slice(0, j).concat(prev.slice(j + 1));
            land[i][j] += Math.max(...targetArr);
        }
    }
    
    return Math.max(...land[land.length - 1]);
}